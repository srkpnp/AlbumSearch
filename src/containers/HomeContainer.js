import React, { Component } from "react";
import Home from '../components/Home';
import { connect } from 'react-redux';
import {increment,decrement,reset,fetchGithubData,updateEntity,updateSearchText} from '../actions/actions'

export class HomeContainer extends Component {

    render() {
        const name = this.props.location && this.props.location.state ? this.props.location.state.userName : ''
        console.log("this.props.---->",this.props);
        return (
            <div>
                <Home userName={name} {...this.props}/>
            </div>
        );
    }
}

export const mapStateToProps = state => {
  console.log("state---->",state);
    return {
      count: state.data.count,
      searchResults:state.data.searchResults,
      selectedEntity:state.data.selectedEntity,
      entity:state.data.entity,
      searchText:state.data.searchText
    };
  };
  
  export const mapDispatchToProps =(dispatch)=>{
    return {
      reset:(data)=>{
        dispatch(reset(data))
      },
      increment:(data)=>{
        dispatch(increment(data))
      },
      decrement:(data)=>{
        dispatch(decrement(data))
      },     
      fetchGithubData:data=>{
        dispatch(fetchGithubData(data))
      },
      updateEntity:data=>{
        dispatch(updateEntity(data))
      },
      updateSearchText:data => {
        dispatch(updateSearchText(data))
      }
    }
  }


  export default connect(mapStateToProps,mapDispatchToProps)(HomeContainer);