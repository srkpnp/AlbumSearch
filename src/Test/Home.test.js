import React from 'react';
import {configure, shallow} from 'enzyme';
import {HomeContainer} from '../containers/HomeContainer';
import * as actions from '../containers/HomeContainer';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter : new Adapter()})
describe( ' HomeContainer', () => {
    let wrapper;

    beforeEach(()=>{
        wrapper = shallow(<HomeContainer/>)
    })

    it('should test mapStateToProps', () => {
        const expectedAction = {
            count: 1,
            searchResults:[],
            selectedEntity:'Album',
            entity:['Album','Music'],
            searchText:'Rahman'
        }

        const state={
            data :{
                count: 1,
                searchResults:[],
                selectedEntity:'Album',
                entity:['Album','Music'],
                searchText:'Rahman'
            }
        }

        expect(actions.mapStateToProps(state)).toEqual(expectedAction)
    })

    it('Should test mapStatetoPRops method', () =>{
        const dispatch =() =>{}
        expect(actions.mapDispatchToProps(dispatch)).toMatchSnapshot()
    })
})