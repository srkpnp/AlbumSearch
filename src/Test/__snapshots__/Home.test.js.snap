// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` HomeContainer Should test mapStatetoPRops method 1`] = `
Object {
  "decrement": [Function],
  "fetchGithubData": [Function],
  "increment": [Function],
  "reset": [Function],
  "updateEntity": [Function],
  "updateSearchText": [Function],
}
`;
